openhtf/core/phase_executor.py

```diff
@@ -303,10 +303,18 @@ def _execute_phase_once(
   ) -> Tuple[PhaseExecutionOutcome, Optional[pstats.Stats]]:
     """Executes the given phase, returning a PhaseExecutionOutcome."""
     # Check this before we create a PhaseState and PhaseRecord.
-    if phase_desc.options.run_if and not phase_desc.options.run_if():
-      self.logger.debug('Phase %s skipped due to run_if returning falsey.',
+    if phase_desc.options.run_if:
+      try:
+        run_phase = phase_desc.options.run_if()
+      except Exception:  # pylint: disable=broad-except
+        # Allow graceful termination
+        return PhaseExecutionOutcome(phase_descriptor.PhaseResult.STOP), None
+
+      if not run_phase:
+        self.logger.debug('Phase %s skipped due to run_if returning falsey.',
```

Comment:
Same as a previous comment, please line up the next line with the opening parenthesis.

Reply:
Done